client/telemetry/src/lib.rs

```diff
@@ -96,14 +96,35 @@ pub struct TelemetryConfig {
 ///
 /// The URL string can be either a URL or a multiaddress.
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
-pub struct TelemetryEndpoints(Vec<(String, u8)>);
+pub struct TelemetryEndpoints(Vec<(Multiaddr, u8)>);
 
 impl TelemetryEndpoints {
-	pub fn new(endpoints: Vec<(String, u8)>) -> Self {
-		TelemetryEndpoints(endpoints)
+	pub fn new(endpoints: Vec<(String, u8)>) -> Result<Self, libp2p::multiaddr::Error> {
+		let endpoints: Result<Vec<(Multiaddr, u8)>, libp2p::multiaddr::Error> = endpoints.iter()
+			.map(|e| Ok((url_to_multiaddr(&e.0)?, e.1)))
+			.collect();
+		endpoints.map(Self)
```

Comment:
I first had a one-liner that was written like this:
```rust
	pub fn new(endpoints: Vec<(String, u8)>) -> Result<Self, libp2p::multiaddr::Error> {
        endpoints.iter()
	        .map(|e| Ok((url_to_multiaddr(&e.0)?, e.1)))
			.collect::<Result<Vec<Multiaddr, u8>, libp2p::multiaddr::Error>>()
		    map(Self)
```
but ultimately decided against it for code clarity.